#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-mysql-real-escape-string.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: systems running on sqlite3 (or pgsql) this function is not available
## DP: instead of changing the whole code, lets hack our own version of this function.
## DP: according to the documentation found here: http://il2.php.net/mysql_real_escape_string
## DP: this shold be enough.
## DP: fixes upstream: http://www.freepbx.org/trac/ticket/1963



@DPATCH@
diff -urNad upstream~/amp_conf/htdocs/admin/header.php upstream/amp_conf/htdocs/admin/header.php
--- upstream~/amp_conf/htdocs/admin/header.php	2006-12-09 01:48:17.000000000 +0200
+++ upstream/amp_conf/htdocs/admin/header.php	2007-05-30 12:31:59.000000000 +0300
@@ -74,6 +74,24 @@
 	textdomain('amp');
 }
 
+// systems running on sqlite3 (or pgsql) this function is not available
+// instead of changing the whole code, lets hack our own version of this function.
+// according to the documentation found here: http://il2.php.net/mysql_real_escape_string
+// this shold be enough.
+if (!function_exists('mysql_real_escape_string')) {
+	function mysql_real_escape_string($str) {
+		$str = str_replace( "\x00", "\\" . "\x00", $str );
+		$str = str_replace( "\x1a", "\\" . "\x1a", $str );
+		$str = str_replace( "\n" , "\\". "\n"    , $str );
+		$str = str_replace( "\r" , "\\". "\r"    , $str );
+		$str = str_replace( "\\" , "\\". "\\"    , $str );
+		$str = str_replace( "'" , "\\". "'"      , $str );
+		$str = str_replace( '"' , "\\". '"'      , $str );
+		return $str;
+	}
+}
+
+
 if (!$quietmode) {
 ?>
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
